Реализовать UI слой экрана Выбор региона

JudjinGM opened the pull request:
Реализовал получение регионов по приходящим аргументам(id страны). Если id == null, то приходит весь список регионов. Сделал фильтрацию по имени региона.

Навигация в этой ветке неверная, сделал для возможности перейти сразу на выбор регионов.

JudjinGM committed:
Реализовал получение регионов по приходчщим аргументам(id страны). Если id == null, то приходит весь списко регионов. Сделал филтрацию по имени региона.

Навигация в этой ветке неверная, сделал для возможности перейти сразу на выбор регинов.

BigMag74 commented on app/src/main/java/ru/practicum/android/diploma/filter/ui/fragment/FilteringAreaFragment.kt:
```diff
@@ -25,8 +45,98 @@ class FilteringAreaFragment : Fragment() {
+    private fun recycleViewInit() {
+        areasAdapter = RVAdapter { item ->
+            if (isClickDebounce()) {
+                viewModel.saveArea(item.id)
```
Насколько я понял сохранять в локалКэш нужно только на экране "Выбор места работы", здесь нужно просто перейти на этот экран и передать туда параметры страны (если она есть), и региона
Можно сделать как-то так
 setFragmentResult(
                REQUEST_KEY,
                bundleOf(BUNDLE_KEY_FOR_AREA to AreaFilter(id, name, countryName))
            )
            findNavController().popBackStack(R.id.filteringChoosingWorkplaceFragment, false)

JudjinGM committed:
Поправил замечания:
1. Клик на весь элемент
2. Навигация и передача area через fragment result api.
